proposals/XXXX-compact.md

```diff
@@ -96,79 +104,167 @@ Compact witness type signatures are a kind of foreign function interface (FFI).
 
 ### Ledger ADTs and public state updates
-Ledger ADTs have both an off-chain and an on-chain representation, and their operations have both off-chain and on-chain implementations.
+Ledger ADTs have both an off-chain and an on-chain data representation, and their operations have both off-chain and on-chain computational implementations.
 In the Midnight Network we have chosen to make both off-chain and on-chain parts the same.
-The Midnight-specific Compact runtime actually uses the same representation as the Midnight Network for ledger ADTs,
-and it actually uses the Impact VM to implement ledger operations.
+The Midnight-specific Compact runtime actually uses the *same* representation as the Midnight Network does for ledger ADTs,
+and it actually uses the Impact VM off chain to implement ledger operations.
 
 However, it is not necessary that Compact uses the same implementation off and on chain.
 The Compact compiler itself simply needs to have a backend-specific implementation of off-chain code to use for the ledger,
-and a backend-specifi implementation of the on-chain code to be deployed in a contract or else submitted in a transaction depending on how the backend blockchain works.
+and a backend-specific implementation of the on-chain code to be deployed in a contract or else submitted in a transaction, depending on how the backend blockchain works.
+
+## Open Language Design
+The open-source Compact repository will host language feature design documents.
+The Midnight Network has a process for proposing changes called the Midnight Improvement Proposal (MIP) process.
+The Compact project will use this process or a suitable variation of it to conduct language feature design in the open,
+and to allow community-driven language feature design.
 
-<mark>_**Effort and resources** committed (coders and any other resources
-that are needed) and timeline._
+An example of a similar process being used successfully for programming languages can be found in the Python language community's
+[Python Enhancement Proposal (PEP)](https://peps.python.org/) process.
+
+<mark>**Effort and resources** committed (coders and any other resources that are needed) and timeline.
 </mark>
 
 # How To
-<mark>_**How to**: How to host and test the project. How to deploy and use.
-How does one know that it works._
+<mark>**How to**: How to host and test the project.
+How to deploy and use.
+How does one know that it works.
 </mark>
 
 # References
-<mark>_**References**. See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html)._
+<mark>**References**. See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html).
 </mark>
 
-# Closure
-<mark>_**Closure** how do we know that the project succeeded. This has to
-be measurable if possible. Make references to successor projects if
-any._
-</mark>
+TODO, suggestions:
```

Comment:
Personally I think referencing Chez Scheme, Nanopass framework, and the long Kachina paper makes sense. We could even specify that Compactc is implemented in Chez Scheme.